pkg/cloudprovider/vsphere/nodemanager_test.go

```diff
@@ -267,3 +268,825 @@ func TestReturnIPsFromSpecificFamily(t *testing.T) {
+				{Type: "ExternalIP", Address: "172.15.108.11"},
+			},
+		},
+		{
+			testName: "BySettingANetworkNameThatDoesntExist",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv4"},
+				cpiConfig: &ccfg.CPIConfig{
+					Nodes: ccfg.Nodes{
+						InternalVMNetworkName: "internal_net",
+						ExternalVMNetworkName: "external_net",
+					},
+				},
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "net_a",
+						IpAddress: []string{
+							"10.10.1.22",
+						},
+					},
+					{
+						Network: "net_b",
+						IpAddress: []string{
+							"172.15.108.11",
+						},
+					},
+				},
+			},
+			expectedErrorSubstring: "unable to find suitable IP address for node",
+		},
+		{
+			testName: "ByDefaultSelection_TheSecondNICHasNoIPs",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv4"},
+				cpiConfig:        nil,
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "net_a",
+						IpAddress: []string{
+							"172.15.108.11",
+						},
+					},
+					{
+						Network:   "net_b",
+						IpAddress: []string{},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "172.15.108.11"},
+				{Type: "ExternalIP", Address: "172.15.108.11"},
+			},
+		},
+		{
+			testName: "ByDefaultSelection_TheFirstNICHasNoIPs",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv4"},
+				cpiConfig:        nil,
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network:   "net_a",
+						IpAddress: []string{},
+					},
+					{
+						Network: "net_b",
+						IpAddress: []string{
+							"172.15.108.11",
+						},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "172.15.108.11"},
+				{Type: "ExternalIP", Address: "172.15.108.11"},
+			},
+		},
+		{
+			testName: "ByDefaultSelection_TheFirstNICHasNoIPsOfTheDesiredFamily",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv4"},
+				cpiConfig:        nil,
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "net_a",
+						IpAddress: []string{
+							"fd00:cccc::1",
+						},
+					},
+					{
+						Network: "net_b",
+						IpAddress: []string{
+							"172.15.108.11",
+						},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "172.15.108.11"},
+				{Type: "ExternalIP", Address: "172.15.108.11"},
+			},
+		},
+		{
+			testName: "ByDefaultSelection_TheSecondNICHasNoIPsOfTheDesiredFamily",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv4"},
+				cpiConfig:        nil,
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "net_a",
+						IpAddress: []string{
+							"172.15.108.11",
+							"fe80:cccc::1",
+						},
+					},
+					{
+						Network: "net_b",
+						IpAddress: []string{
+							"fe80:cccc::2",
+						},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "172.15.108.11"},
+				{Type: "ExternalIP", Address: "172.15.108.11"},
+			},
```

Comment:
We believe this is a scenario where a prior (and reverted) dualstack PR caused an undesirable change in behavior.